### DISCUSSION TITLE
Where is the ~/projects directory located?

### DISCUSSION THREAD

**Discussion by rarygit** [idaholab/moose#19044] [Q&A Getting Started] [answered]
I am following the online installation guide.
I am having difficulty with the last step:  ./run_tests -j 4

The error message: ModuleNotFoundError: No module named 'TestHarness'

This might be due to my confusion where MOOSE_DIR and FRAMEWORK_DIR are located.
For me, the installation guide is clear how to install in conda, but not its relationship to the projects dir

OS: Ubuntu 20.04  
Conda MOOSE Environment
https://mooseframework.inl.gov/getting_started/installation/conda.html
This is OK so far:
conda create --name moose -q -y
conda activate moose
conda install moose-tools
conda install moose-libmesh

Obtaining and Building MOOSE
https://mooseframework.inl.gov/getting_started/installation/install_moose.html
This is OK so far:
mkdir ~/projects    Where are you meant to make the /projects folder?
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
cd ~/projects/moose/test
make -j 4

But not this:
./run_tests -j 4

First question:
At this point, I need clarity here regarding my moose and project folders 
Is the projects dir in my moose conda env?
Or is it located somewhere else, even another HDD?

Is this path to projects correct?
~/conda_envs/envs/moose/projects

and is this the  path to framework?
~/conda_envs/envs/moose/projects/moose/framework

Therefore: 
MOOSE_DIR = ~/conda_envs/envs/moose/
and
FRAMEWORK_DIR = ~/conda_envs/envs/moose/projects/moose/framework

+++++++++++++++++++++++++++++++++++++

Second question:
In Troubleshooting, you use /tmp folder
Where is /tmp in relation to ~/conda_env/env/moose ? 

https://mooseframework.inl.gov/help/troubleshooting.html#buildissues
Try building a simple hello world example (there is more text than what is visible, be sure to copy it all):
cd /tmp
cat << EOF > hello.C

Thanks for any help or suggestions

**Comment by rarygit** [comment]
From this post:
https://github.com/idaholab/moose/discussions/18902
https://user-images.githubusercontent.com/55847507/134444152-d84a78d1-ff4d-412f-97b2-64c3cd4ab784.png

I see that the MOOSE_DIR is in the conda_env,
and FRAMEWORK_DIR is elsewhere, in this case the USER dir on mac OSX

on Ubuntu OS:
MOOSE_DIR = ~/conda_envs/envs/moose/
and
FRAMEWORK_DIR = /media/USER/projects/moose/framework
alternatively,
FRAMEWORK_DIR = /media/USER/some other HDD/projects/moose/framework 

Is that now correct?

**Comment by loganharbour** [comment]
MOOSE_DIR should be the path to where you _cloned_ moose into. It has nothing to do with conda. If you ran make within the tests directory, MOOSE_DIR should be one level up from this.

**Comment by rarygit** [comment]
@loganharbour 

Thanks for replying and clarifying the paths.
I am reinstalling and trying again.

Which page on Moose (mooseframework.inl.gov) has the link to this url:  "Obtaining and Building MOOSE"
https://mooseframework.inl.gov/getting_started/installation/install_moose.html

Installation page has links to install the MOOSE Framework on Linux and macOSX, 
https://mooseframework.inl.gov/getting_started/installation/

which lands you on the Install conda moose environment. 
https://mooseframework.inl.gov/getting_started/installation/conda.html

This page advises the user to install, compile and test MOOSE by running the following commands in a terminal.
mkdir ~/projects
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
cd ~/projects/moose/test
make -j 4
./run_tests -j 4

That ends in failure with errors.

In fact there is another web page called "Obtaining and Building MOOSE", which is absent from the Installation or Conda pages
https://mooseframework.inl.gov/getting_started/installation/install_moose.html

I stumbled upon on it while googling searches about /projects and setting up the moose framework
The commands on that page are:

"To install MOOSE run the following commands in a terminal."
mkdir ~/projects
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
cd ~/projects/moose
unset PETSC_DIR PETSC_ARCH
./scripts/update_and_rebuild_petsc.sh
./scripts/update_and_rebuild_libmesh.sh
cd ~/projects/moose/test
make -j 4
./run_tests -j 4

Which are considerably different commands to those provided on the conda moose environment page.

I will try this, using:
MOOSE_DIR = the path to where you cloned moose into. MOOSE_DIR is one level up from the test directory. 
FRAMEWORK_DIR = MOOSE_DIR/framework

**Comment by rarygit** [comment, marked as answer]
Ok, that worked well!

Set MOOSE_DIR and FRAMEWORK_DIR in both .bashrc and the Makefile in /projects/moose/test
Given the time it takes to get to finally ./run_tests, I wasn't taking any risks.

.....
./run_tests -j 4
Ran 2912 tests in 832.8 seconds. Average test time 0.9 seconds, maximum test time 48.9 seconds.
2912 passed, 96 skipped, 0 pending, 0 failed

All up, the cloning, installing, compiling, testing of moose on "/media/hDD/projects/moose" swallowed 11.4 Gb,
and took a few hours to complete.

Thanks to @loganharbour, problem solved.
